pandas/io/excel.py

```diff
@@ -225,6 +225,9 @@ def read_excel(io, sheet_name=0, header=0, skiprows=None, skip_footer=0,
     elif 'sheetname' in kwds:
         raise TypeError("Cannot specify both `sheet_name` and `sheetname`. "
                         "Use just `sheet_name`")
+    if 'sheet' in kwds:
+        raise TypeError("read_excel() got an unexpected keyword argument "
+                        "`sheet`")
```

Comment:
This is not generalizable.  What happens if I were to pass in an argument called "Sheet" ?  Your check wouldn't catch that.  You should develop a more general checker that checks for **any** keywords that don't correspond to ones that we accept.